cli/ops/files.rs

```diff
@@ -129,17 +124,13 @@ pub fn op_seek(
 ) -> Result<JsonOp, ErrBox> {
   let args: SeekArgs = serde_json::from_value(args)?;
 
-  match resources::lookup(args.rid as u32) {
-    None => Err(deno_error::bad_resource()),
-    Some(resource) => {
-      let op = resources::seek(resource, args.offset, args.whence as u32)
-        .and_then(move |_| futures::future::ok(json!({})));
-      if args.promise_id.is_none() {
-        let buf = op.wait()?;
-        Ok(JsonOp::Sync(buf))
-      } else {
-        Ok(JsonOp::Async(Box::new(op)))
-      }
-    }
+  let resource = resources::lookup(args.rid as u32)?;
```

Comment:
If it's a bad rid, you should get a BadResource error from `lookup()`

Reply:
Alright, but then again inside `poll_read` and `poll_write` if `rid` is not present in resource table we'd return different error type (`std::io::Error` because `AsyncRead/Write` requires it) than during lookup.